### ISSUE TITLE
md-input

### ISSUE THREAD

**Issue by jelbourn** [angular/material2#116]
Blocked on #84

**Comment by ekarlso** [comment]
Hi, is there any eta on this part?

**Comment by jelbourn** [comment]
@ekarlso it's in progress #161

**Comment by hansl** [comment]
Merged #161 . I'm going to close this and open issues for missing parts.

**Comment by fxck** [comment]
@hansl @jelbourn I can tell you about few

- no styles for invalid inputs
- there's an issue with using `[value]` instead of `[ngModel]`, but that could be related to https://github.com/angular/angular/issues/7690
- no textareas
- no bottom margin, this is way too stacked on each other 
![image](https://cloud.githubusercontent.com/assets/1303561/14359135/6d6ead2e-fcf0-11e5-8573-b1bf3938be87.png)
- while I like that there's just one component, unlike ng1 which had wrapper `<md-input-container>`, it makes it less friendly for customisation, since you can't easily utilise the `md-is-active` css class

**Comment by jelbourn** [comment]
@fxck 
* Invalid styles and textareas will be coming
* The lack of margins is an intentional, library-wide design decision. It should be up to the user of the components to set margins, not the authors. 
* I'm not sure what you're referring to with `md-is-active`; I don't see a reference to this anywhere in the material1 source. Do you have some customization use-case in mind?

**Comment by fxck** [comment]
> The lack of margins is an intentional, library-wide design decision. It should be up to the user of the components to set margins, not the authors.

Good to know, thanks! Although material1 used that free space to show errors, which seems reasonable.

> I'm not sure what you're referring to with md-is-active; I don't see a reference to this anywhere in the material1 source. Do you have some customization use-case in mind?

Yea, sorry, I meant !md-input-focused`, basically I want this...

![image](https://cloud.githubusercontent.com/assets/1303561/14359941/24c652c6-fcf4-11e5-91fc-1957e476674e.png)

![image](https://cloud.githubusercontent.com/assets/1303561/14359948/2c428aa6-fcf4-11e5-91b0-2222ad1c346e.png)

...which means my own "pseudo-label", and I want it to be blue when the input is focused, I haven't used material1 for ages, but think this was possible

```html
<md-input-container>
  <h2 class="my-special-label">Whatever</h2>    
  <input ng-model="foo.bar">
</md-input-container>
```

```css
.md-input-focused .my-special-label {
  color: whatever;
}
```

**Comment by jelbourn** [comment]
@fxck we could potentially add an `isFocused` property to the input. That would enable:
```html
<md-input #nickname [class.pajama-party]="nickname.isFocused">
```

**Comment by fxck** [comment]
Yeah that's what I was thinking as well. That would solve the problem.

**Comment by hansl** [comment]
There is already, it’s called `focused`.

-- 
Hans Larsen
Sent with Airmail

On April 7, 2016 at 10:29:49 AM, Aleš (notifications@github.com) wrote:

Yeah that's what I was thinking as well. That would solve the problem.

**Comment by fxck** [comment]
Hello there @jelbourn @hansl, are those textareas and error styles going to be part of alpha.3? Since that new issue for missing parts doesn't really exist yet, it's hard to guess. Thanks!

**Comment by jelbourn** [comment]
@fxck we haven't scheduled those features yet- the team is gearing up for ng-conf, so it will be some time after that.

**Comment by fxck** [comment]
I know I'm annoying, but is there ETA for textareas yet? I wonder by what method you decide which components have bigger priority than other, because to me, it would make sense to start with the basics, to which textarea arguably belongs, and then follow up with more fancy stuff, like `md-bottom-nav`..

**Comment by jelbourn** [comment]
@fxck No ETA yet; one of our 20% contributors will probably take this on after they're finished with their current stuff.

**Comment by AhsanAyaz** [comment]
+1 for the text area

**Comment by CoreyGrant** [comment]
Are there any plans on validation?
